lcls_tools/common/devices/wire.py

```diff
@@ -0,0 +1,281 @@
+from datetime import datetime
+from functools import wraps
+from pydantic import (
+    BaseModel,
+    PositiveFloat,
+    SerializeAsAny,
+    field_validator,
+)
+from typing import (
+    Dict,
+    List,
+    Optional,
+    Union,
+)
+from lcls_tools.common.devices.device import (
+    Device,
+    ControlInformation,
+    Metadata,
+    PVSet,
+)
+from epics import PV
+
+EPICS_ERROR_MESSAGE = "Not able to connect to the control system."
+
+class WirePVSet(PVSet):
+    motr: PV # the rest of the PVs are all related to the motor
+    cnen: PV
+    velo: PV # velocity
+    rbv: PV
+    rmp: PV  # retracted motor position?
+    init: PV
+    retract: PV
+    startscan: PV
+    usexwire: PV
+    useywire: PV
+    useuwire: PV
+    xwireinner: PV
+    xwireouter: PV
+    ywireinner: PV
+    ywireouter: PV
+    uwireinner: PV
+    uwireouter: PV
+    enabled: PV
+    homed: PV
+    timeout: PV
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    @field_validator("*", mode="before")
+    def validate_pv_fields(cls, v: str) -> PV:
+        return PV(v)
+
+
+class WireControlInformation(ControlInformation):
+    PVs: SerializeAsAny[WirePVSet]
+    _ctrl_options: SerializeAsAny[Optional[Dict[str, int]]] = dict()
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        #TODO: Get possible options for wire ctrl PV
+        options = self.PVs.motr.get_ctrlvars()["enum_strs"]
+        [self._ctrl_options.update({option: i}) for i, option in enumerate(options)]
+
+    @property
+    def ctrl_options(self):
+        return self._ctrl_options
+
+
+class WireMetadata(Metadata):
+    thickness: Optional[PositiveFloat] = None
+    speed: Optional[PositiveFloat] = None
+    xsize: Optional[PositiveFloat] = None
+    ysize: Optional[PositiveFloat] = None
+    usize: Optional[PositiveFloat] = None
+    #TODO: Add wire material and sum_l here?
+    #TODO: Add LBLM and BPM infomration here? 
+    #TODO: Add info on locations for X, Y, U wires
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+
+class Wire(Device):
+    controls_information: SerializeAsAny[WireControlInformation]
+    metadata: SerializeAsAny[WireMetadata]
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    """ Decorators """
+
+    def check_state(f):
+        """Decorator to only allow transitions in 'Ready' state"""
+
+        def decorated(self, *args, **kwargs):
+            #TODO: Find out right check for this spot/wires
+            if self.motr != "Ready":
+                print("Unable to perform action, wire not in Ready state")
```

Comment:
I am actually not sure. I agree that it would be very useful to send the base device name around for error handling purposes.

@nneveu Are you aware of a method to grab the base name?

Reply:
The PV names should be initialed by this point and available through the PVSet. Is that what you mean? We can discuss in the PR meeting.

Reply:
Add this as an issue later